src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+const VALID_CHARS
+= 'qwertyuiopasdfghjklzxcvbnmйцукенгшщзхїфівапролджєячсмитьбю1234567890 ';
+
+const getUserById = (id: number) => {
+  return usersFromServer.find(user => user.id === id);
+};
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [authorId, setAuthorId] = useState(0);
+  const [preparedTodos, setPreparedTodos] = useState(todos);
+  const [isAuthorIdClicked, setIsAuthotIdClicked] = useState(false);
+  const [isTitleClicked, setIsTitleClicked] = useState(false);
+
+  const getNextId = () => {
+    const maxId = Math.max(...preparedTodos.map(({ id }) => id));
+
+    return maxId + 1;
+  };
```

Comment:
please move this function outside of the component, don't need it to be re-rendered every time the component is updated